### PR TITLE
[BEAM-6904] Add tests for structuralValue implementation in coders

### PR DESCRIPTION
**Pull request by AlexKbit**
Hi @kennknowles !
This change contains tests on structuralValue implementation for base coders (with deprecate consistentWithEquals)

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python3_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) <br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by AlexKbit** [comment]
retest this please

**Comment by AlexKbit** [comment]
@kennknowles Could you please review this changes?

File: sdks/java/core/src/test/java/org/apache/beam/sdk/coders/BigIntegerCoderTest.java [hunk 1]
```diff
@@ -96,4 +96,19 @@ public void testStructuralValueReturnTheSameValue() {
...
+
+  @Test
+  public void testStructuralValueDecodeEncodeEqual() throws Exception {
+    for (BigInteger value : TEST_VALUES) {
+      CoderProperties.structuralValueDecodeEncodeEqual(TEST_CODER, value);
+    }
+  }
+
+  @Test
+  public void testStructuralValueConsistentWithEquals() throws Exception {
```
Comment (kennknowles) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/coders/BigIntegerCoderTest.java]: For `structuralValueConsistentWithEquals` it is also good to check non-equal members. So the test at `BitSetCoderTest.testStructuralValueConsistentWithEquals` is a good example of how to use it.

Comment (AlexKbit) [reply sdks/java/core/src/test/java/org/apache/beam/sdk/coders/BigIntegerCoderTest.java]: Ok, I'll change this.

**Comment by AlexKbit** [comment]
retest this please

**Comment by AlexKbit** [comment]
retest this please

**Comment by AlexKbit** [comment]
retest this please

File: sdks/java/core/src/test/java/org/apache/beam/sdk/coders/BigIntegerCoderTest.java (hunk 1 above)
Comment (AlexKbit) [reply sdks/java/core/src/test/java/org/apache/beam/sdk/coders/BigIntegerCoderTest.java]: Updated

**Comment by AlexKbit** [comment]
@kennknowles I added checks for non-equal members

**Comment by lukecwik** [comment]
@kennknowles Whats the status on this?

**Comment by kennknowles** [comment]
Ah, my mistake. I forgot to actually "request" my own review so it was not showing in my review queue.

**Comment by kennknowles** [comment]
These tests are definitely useful. I want to merge it. I will take a minute and see if I can resolve the conflicts. Since it is caused by my delay, please allow me and I will push back to this branch if I have perms.

**Comment by kennknowles** [comment]
Ah, and the conflicts are caused by a prior commit being reverted because of issues with Dataflow pipeline update.

**Comment by kennknowles** [comment]
Indeed, we need to roll forward #8258 for this one.

**Comment by AlexKbit** [comment]
I will can update this PR (solve conflicts).
But Kenn is right, we should merge my previous changes # 8071 which was reverted in # 8258

**Comment by kennknowles** [comment]
I don't want to waste you time, @AlexKBit. It seems that a lot of these tests are for methods that were rolled back. So if you want to submit just the tests that apply to existing implementations, that's cool. But maybe you would prefer to wait. The issue with update incompatibility in Dataflow needs to get resolved anyhow.

**Comment by AlexKbit** [comment]
Ok I think I can wait or add patch to ticket in Jira. 
@kennknowles Will you close this PR?

**Comment by kennknowles** [comment]
It is OK with me you can leave this open or close it. Might as well keep the useful code waiting until the right time.

**Pull request closed without merging**